core/network/src/test/sync.rs

```diff
@@ -110,6 +110,23 @@ fn sync_after_fork_works() {
 	assert!(net.peer(2).client.backend().blockchain().canon_equals_to(&peer1_chain));
 }
 
+#[test]
+fn syncs_all_forks() {
+	::env_logger::init().ok();
+	let mut net = TestNet::new(4);
+	net.sync_step();
+	net.peer(0).push_blocks(2, false);
+	net.peer(1).push_blocks(2, false);
+
+	net.peer(0).push_blocks(2, true);
+	net.peer(1).push_blocks(4, false);
+
+	net.sync();
+	// Check that all peers have all of the blocks.
+	assert_eq!(9, net.peer(0).client.backend().blockchain().blocks_count());
```

Comment:
Why is the number 9 when you pushed `10` blocks, do I miss something?
And aren't you creating 4 peers, so shouldn't you test all peers? (like said in the comment)

Reply:
each peer starts with genesis block,
before sync peer 0 has 5 blocks, peer 1 has 7. 3 of these are the same, so each peer ends up with 9.